src/views/Home.jsx

```diff
@@ -17,15 +20,52 @@ export function Home({ createToken, setListToken }) {
 		} else {
 			listId = null;
 			setListToken(listId);
-			setMessage('Your shopping list was not created. Please try again. ');
+			setCreateListMessage(
+				'Your shopping list was not created. Please try again. ',
+			);
 		}
 	}
 
+	async function handleJoinClick(e) {
+		e.preventDefault();
+
+		if (!tokenInput) {
+			setExistingListMessage('Please enter a token.');
+		}
+		const listExists = await checkIfListExists(tokenInput);
+		if (listExists) {
+			setListToken(tokenInput);
+			navigate('/list');
+		} else {
+			setExistingListMessage(' Enter a valid token or create a new list.');
+			setTokenInput('');
+		}
+	}
+	function handleTokenInputChange(e) {
+		setTokenInput(e.target.value);
+	}
+
 	return (
 		<div className="Home">
 			<h2>Welcome to your Smart Shopping List</h2>
-			<button onClick={handleClick}>Create a new list</button>
-			<p>{message}</p>
+			<form onSubmit={handleJoinClick}>
```

Comment:
Let's change this function name to `handleSubmit`, `handleTokenInputFormSubmit` or something between those lines.

Even though `handleJoinClick` is a good name, it implies that the form will only have ONE and only ONE button. If we decide that we need to have another button inside the form someday in the future, then that method name will lose its meaning, and we will need to change it to reflect that.

Reply:
Good point. We implemented and pushed this change.